src/gt4py/next/program_processors/runners/dace_iterator/__init__.py

```diff
@@ -155,214 +147,3 @@ def build_sdfg_from_itir(
-            self.itir,
-            arg_types,
-            offset_provider=offset_provider,
-            column_axis=kwargs.get("column_axis", None),
-        )
-        self.sdfg_closure_vars["sdfg.arrays"] = sdfg.arrays  # use it in __sdfg_closure__
-
-        # Halo exchange related metadata, i.e. gt4py_program_input_fields, gt4py_program_output_fields, offset_providers_per_input_field
-        # Add them as dynamic properties to the SDFG
-
-        assert all(
-            isinstance(in_field, SymRef)
-            for closure in self.itir.closures
-            for in_field in closure.inputs
-        )  # backend only supports SymRef inputs, not `index` calls
-        input_fields = [
-            str(in_field.id)  # type: ignore[union-attr]  # ensured by assert
-            for closure in self.itir.closures
-            for in_field in closure.inputs
-            if str(in_field.id) in fields  # type: ignore[union-attr]  # ensured by assert
-        ]
-        sdfg.gt4py_program_input_fields = {
-            in_field: dim
-            for in_field in input_fields
-            for dim in fields[in_field].dims  # type: ignore[union-attr]
-            if dim.kind == common.DimensionKind.HORIZONTAL
-        }
-
-        output_fields = []
-        for closure in self.itir.closures:
-            output = closure.output
-            if isinstance(output, itir.SymRef):
-                if str(output.id) in fields:
-                    output_fields.append(str(output.id))
-            else:
-                for arg in output.args:
-                    if str(arg.id) in fields:  # type: ignore[attr-defined]
-                        output_fields.append(str(arg.id))  # type: ignore[attr-defined]
-        sdfg.gt4py_program_output_fields = {
-            output: dim
-            for output in output_fields
-            for dim in fields[output].dims  # type: ignore[union-attr]
-            if dim.kind == common.DimensionKind.HORIZONTAL
-        }
-
-        sdfg.offset_providers_per_input_field = {}
-        itir_tmp = legacy_itir_transforms.apply_common_transforms(
-            self.itir, offset_provider=offset_provider
-        )
-        itir_tmp_fencil = program_to_fencil.program_to_fencil(itir_tmp)
-        for closure in itir_tmp_fencil.closures:
-            params_shifts = itir_transforms.trace_shifts.trace_stencil(
-                closure.stencil, num_args=len(closure.inputs)
-            )
-            for param, shifts in zip(closure.inputs, params_shifts):
-                assert isinstance(
-                    param, SymRef
-                )  # backend only supports SymRef inputs, not `index` calls
-                if not isinstance(param.id, str):
-                    continue
-                if param.id not in sdfg.gt4py_program_input_fields:
-                    continue
-                sdfg.offset_providers_per_input_field.setdefault(param.id, []).extend(list(shifts))
```

Comment:
Same here.

All these properties are needed for the automatic halo exchange placement.

Reply:
Results from discussion in a separate channel for "offset_providers_per_input_field":
- it is correct, as only used in `icon4py` which uses "unstructured" fields. There is max one horizontal dimension per field.
- trace_shifts should work™️ partially on GTIR (but not accross `as_fieldop`).
- currently not actively used in `icon4py`, so defer implementation and put a TODO instead.